flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/HybridSourceSplitEnumerator.java

```diff
@@ -246,7 +246,9 @@ public void handleSourceEvent(int subtaskId, SourceEvent sourceEvent) {
 
     @Override
     public void close() throws IOException {
-        currentEnumerator.close();
+        if (currentEnumerator != null) {
```

Comment:
@mbenenso can you please add a comment why it is necessary to do the null check here?

Reply:
[~thw] comment added.